src/Provider.php

```diff
@@ -184,13 +186,15 @@ public function delete_tasks_on_action_deletion( int $action_id ): void {
 
 		$task_ids = implode( ',', array_unique( array_map( 'intval', $task_ids ) ) );
 
-		DB::query(
-			DB::prepare(
-				"DELETE FROM %i WHERE %i IN ({$task_ids})",
-				Task_Logs::table_name(),
-				'task_id',
-			)
-		);
+		if ( $this->container->get( Logger::class ) instanceof DB_Logger ) {
```

Comment:
Here you're using the class, `DB_Logger` to know whether it's using the table to delete logs from.
But you've update, in this PR, the `Logger` interface to expose a `uses_own_table()` method, why not use that to know whether to delete logs from the table or not?
This would make this code future-proof if other loggers are used that use that table and maybe other solutions.
You've already built the logger at this point, so the check is not adding much overhead.